yt_dlp/extractor/flextv.py

```diff
@@ -0,0 +1,50 @@
+from .common import InfoExtractor
+from ..networking.exceptions import HTTPError
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    parse_iso8601,
+    str_or_none,
+    traverse_obj,
+    url_or_none,
+    UserNotLive,
+)
+
+
+class FlexTVIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?flextv\.co\.kr/channels/(?P<id>\d+)/live'
+    _TESTS = [{
+        'url': 'https://www.flextv.co.kr/channels/570421/live',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.flextv.co.kr/channels/746/live',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        channel_id = self._match_id(url)
+
+        try:
+            stream_data = self._download_json(
+                f'https://api.flextv.co.kr/api/channels/{channel_id}/stream?option=all', channel_id)
+        except ExtractorError as e:
+            if isinstance(e.cause, HTTPError) and e.cause.status == 400:
+                raise UserNotLive(video_id=channel_id)
+
+        playlist_url = traverse_obj(stream_data, ('sources', 0, 'url'))
```

Comment:
this is ok to be fatal
```suggestion
        playlist_url = stream_data['sources'][0]['url']
```